src/contract_wrappers/exchange_wrapper.ts

```diff
@@ -565,13 +600,21 @@ export class ExchangeWrapper extends ContractWrapper {
         return eventEmitter;
     }
     /**
-     * Returns the ethereum address of the current exchange contract
+     * Returns the ethereum addresses of all available exchange contracts
      * on the network that the provided web3 instance is connected to
-     * @return  The ethereum address of the current exchange contract.
+     * @return  The ethereum addresses of all available exchange contract.
      */
-    public async getContractAddressAsync(): Promise<string> {
-       const exchangeContract = await this._getExchangeContractAsync();
-       return exchangeContract.address;
+    public async getAvailableContractAddressedAsync(): Promise<string[]> {
+        const networkId = await this._web3Wrapper.getNetworkIdIfExistsAsync();
+        if (_.isUndefined(networkId)) {
+            return [];
+        } else {
+            const exchangeAddresses = _.map(
+                _.values(ExchangeArtifactsByName),
+                exchangeArfifact => exchangeArfifact.networks[networkId].address,
```

Comment:
Won't this fail if there are no artifacts for the networkId? `.address` on undefined? You need to check first that `networks[networkId]` exists...